Source/WebCore/rendering/EventRegion.cpp

```diff
@@ -313,18 +317,38 @@ void EventRegionContext::shrinkWrapInteractionRegions()
             auto hint = m_interactionRectsAndContentHints.get(rectForTracking);
             if (hint != region.contentHint)
                 toAddAfterMerge.append(discoveredRegion);
-            else if (growth > std::numeric_limits<float>::epsilon())
-                discoveredRects.append(rect);
+            else if (growth > std::numeric_limits<float>::epsilon()) {
+                // If the discovered region's shape should not be a rounded-rect
+                // with uniform corner radii, its clipPath will be non-empty.
+                if (auto clipPath = discoveredRegion.clipPath) {
+                    AffineTransform transform;
+                    transform.translate(discoveredRegion.rectInLayerCoordinates.location());
+
+                    Path foundPath = *clipPath;
+                    foundPath.transform(transform);
+
+                    discoveredClipPaths.append(foundPath);
+                } else
+                    discoveredRects.append(rect);
+            }
         }
 
         if (canUseSingleRect)
             region.rectInLayerCoordinates = layerBounds;
         else {
-            Path path = PathUtilities::pathWithShrinkWrappedRects(discoveredRects, region.cornerRadius);
-            region.rectInLayerCoordinates = layerBounds;
+            AffineTransform transform;
+            Path shrinkWrappedRects = PathUtilities::pathWithShrinkWrappedRects(discoveredRects, region.cornerRadius);
+
+            Path path;
+            path.addPath(shrinkWrappedRects, transform);
+            for (Path clipPath : discoveredClipPaths)
+                path.addPath(clipPath, transform);
```

Comment:
Nit: you could get away with r-value `AffineTransform`s, i.e. `addPath(rect, { })`